services/authorization/authorization-impl/src/main/java/com/alexkoveckiy/authorization/impl/router/AuthorizationRouter.java

```diff
@@ -1,23 +1,19 @@
 package com.alexkoveckiy.authorization.impl.router;
 
-import com.alexkoveckiy.authorization.api.router.AuthorizationRouterHandler;
-import com.alexkoveckiy.common.protocol.Request;
 import com.alexkoveckiy.common.router.api.AbstractRouterHandler;
+import com.alexkoveckiy.common.router.api.CommonRouterHandler;
+import com.alexkoveckiy.common.router.api.RequestHandler;
 import org.springframework.stereotype.Component;
 
 /**
  * Created by alex on 23.02.17.
  */
 
 @Component
-public class AuthorizationRouter extends AbstractRouterHandler<AuthorizationRouterHandler> {
+public class AuthorizationRouter extends AbstractRouterHandler<RequestHandler> implements CommonRouterHandler {
```

Comment:
When i change it to more concrete like: extends AbstractRouterHandler<AuthorizastionRequestHandler>, then Spring says that there is 2 autowire candidate.